sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/Plan.java

```diff
@@ -21,6 +22,8 @@
  * list of one or more error messages to be shown to the user,
  */
 public interface Plan extends ParentElement<Phase> {
+    void setErrors(List<String> errors);
```

Comment:
I agree to a degree.  I didn't really like adding the `setter` method.  However the point of the implementation is that we have rejected the proposed target and kept the previous good target.  By the lights of the previous target the plan remains unchanged and is correct.  In fact, if an attempt were made to change from a "good" config to a "bad" one, that original good plan would still be executed, or should be.

If anything we should consider putting errors outside the plan interface altogether.  However the implementation here is a return to the original behavior which had regressed.  Because configuration validation and plan generation are so decoupled (as they should be), I didn't see a good solution that avoided the setter.